src/memo/memo.ml

```diff
@@ -741,15 +736,21 @@ let dep_node (type i o f) (t : (i, o, f) t) inp =
     Store.set t.cache inp dep_node;
     dep_node
 
-module Prev_cycle_cache_lookup_result = struct
-  type 'a invalid_value =
-    | No_value
-    | Old_value of 'a Cached_value.t
-    | Cycle_error of Cycle_error.t
+(* When looking up a value cached in a previous run, there can be three errors:
 
+   - [Not_found]: either the value has never been computed before, or the last
+   computation attempt failed.
+
+   - [Out_of_date]: we found a value computed in a previous run but it is out of
+   date; we return that old value so that it can be compared with a new one to
+   support the early cutoff.
+
+   - [Cancelled _]: the cache lookup was cancelled due to a dependency cycle. *)
```

Comment:
Maybe here it is worth clarifying that the dependency cycle was introduced this cycle, not in the previous cycle. Like this?

```
- [Cancelled _]: the cache lookup was cancelled due to a dependency cycle.
This is returned when a new dependency cycle is introduced. If a dependency cycle existed on a previous run, we'll see [Not_found], instead.
```

(assuming that's even true)

Reply:
How about s/it is out of date/it is out of date because one of its dependencies changed/?
(we have two viable contepts of being out of date in theory and that clarifies which one we mean)

Reply:
Thanks, agreed with all suggestions.